freedomofpress/securedrop-workstation
# Harden sd-viewer configuration and provision mailcap default rules; bump version

## Status

Ready for review

## Description of Changes

- Prevent sd-viewer from launching its own disposable VMs (enforced via `default_dispvm` setting)
- Set default `qubes.GetImageRGBA` policy for all workstation VMs (it otherwise can be used to launch disposable VMs)
- Provision default mailcap rules introduced in https://github.com/freedomofpress/securedrop-debian-packaging/pull/221 to AppVMs
  - This (and only this) change requires a full `sdw-admin --apply` run, which is enforced in postinst by dropping a migration flag (interpreted via https://github.com/freedomofpress/securedrop-workstation/blob/main/dom0/securedrop-check-migration). This assumes the package is updated via the graphical updater. (Commit by @conorsch.)
- Bumps version to 0.5.3

## Side effects of RPC policy change

There should not be any negative impact from restricting the `GetImageRGBA` policy across SecureDrop Workstation VMs. `qvm-convert-img` will no longer work in, e.g., `sd-devices`, but `qvm-convert-pdf` already does not work due to our existing `qubes.PdfConvert` policy. If we want to support PDF/image conversion in disposable VMs in future, that will require its own configuration.

## Test plan

Note: Mailcap configuration testing is handled in https://github.com/freedomofpress/securedrop-debian-packaging/pull/221, which is dependent on this PR, and therefore not covered below.

1. Ensure your SecureDrop Workstation environment is provisioned to Qubes 4.0.4 (up-to-date 4.0.3 system is equivalent to 4.0.4) running a staging or production build without custom modifications.
2. Open your development VM and check out this PR's branch.
3. In your `securedrop-workstation` checkout in `dom0`, run `make clone` to build and fetch a fresh RPM.
4. In your `dom0` checkout, enter the directory `rpm-build/RPMS/noarch`. There should be a 0.5.3 RPM. Install it via `sudo dnf install securedrop-workstation-dom0-config-0.5.3-1.fc25.noarch.rpm`. 
5. Force a graphical updater run with `/opt/securedrop/launcher/sdw-launcher.py --skip-delta 0` in `dom0`. 
6. Open any file in a disposable VM using the SecureDrop Client. While the disposable VM is running, open a terminal inside of it using the domain manager.
7. - [ ] Observe that the file opens without problems.
8. In the terminal, type `qrexec-client-vm ‘@dispvm’ qubes.GetImageRGBA < /dev/urandom`
9. Observe that you see the output `Request refused` and that no disposable VM is opened.
10. In `dom0`, run `qvm-prefs sd-viewer default_dispvm`.
11. - [ ] Observe empty output (_not_ `sd-viewer`)
12. In your `securedrop-workstation` directory, ensure that `config.json` is present, then run `make test`
13. - [ ] Observe that the newly added RPC policy test and the `test_sd_viewer_config`  are passing.
14.  Launch a terminal in `sd-log`
15. - [ ] Observe that `qrexec-client-vm ‘@dispvm’ qubes.GetImageRGBA < /dev/urandom` does not open a disposable VM.
